numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -621,6 +594,33 @@ static PyObject *
 /**end repeat**/
 
 static PyObject *
+voidtype_str(PyObject *self)
+{
+    if (PyDataType_HASFIELDS(((PyVoidScalarObject*)self)->descr)) {
+        PyObject *arr, *ret = NULL;
+
+        arr = PyArray_FromScalar(self, NULL);
+        if (arr != NULL) {
+            ret = PyObject_Str((PyObject *)arr);
+            Py_DECREF(arr);
+        }
+        return ret;
```

Comment:
Can we get rid of this, and make a call to something in `core.multiarray._internal`? This is the recursive point I'd rather not have.